src/Psy/Command/ParseCommand.php

```diff
@@ -126,7 +127,9 @@ private function parse($code)
     private function getParser()
     {
         if (!isset($this->parser)) {
-            $this->parser = new Parser(new Lexer());
+            $this->parser = (new ParserFactory())->create(
+                version_compare(PHP_VERSION, '7.0', '>=') ? ParserFactory::ONLY_PHP7 : ParserFactory::ONLY_PHP5
```

Comment:
In the `CodeCleaner` strict options should be used, because parsed code will be evaluated on the same interpreter (so with `PREFER_PHP7` invalid code can be passed to 5.x interpreter).

I don't have strong opinion about `ParseCommand`. I think the best way will be to add additional console option for PHP version. What do you think?